packages/asset-service/src/generateAssetData/generateAssetData.ts

```diff
@@ -33,35 +34,56 @@ const generateAssetData = async () => {
     ...osmosisAssets,
     ...avalancheAssets,
     ...optimismAssets,
+    ...bnbsmartchainAssets,
   ]
+
   // remove blacklisted assets
   const filteredAssetData = filterOutBlacklistedAssets(unfilteredAssetData)
 
-  // For coins not currently in the color map, check to see if we can generate a color from the icon
-  const filteredWithColors = await setColors(filteredAssetData)
-
   // deterministic order so diffs are readable
-  const orderedAssetList = orderBy(filteredWithColors, 'assetId')
+  const orderedAssetList = orderBy(filteredAssetData, 'assetId')
 
   const ethAssetNames = ethAssets.map((asset) => asset.name)
   const avalancheAssetNames = avalancheAssets.map((asset) => asset.name)
   const optimismAssetNames = optimismAssets.map((asset) => asset.name)
+  const bnbsmartchainAssetNames = bnbsmartchainAssets.map((asset) => asset.name)
 
   const generatedAssetData = orderedAssetList.reduce<AssetsById>((acc, asset) => {
     const { chainReference } = fromAssetId(asset.assetId)
 
+    // mark any ethereum assets that also exist on other evm chains
+    if (
+      chainReference === CHAIN_REFERENCE.EthereumMainnet &&
+      asset.assetId !== ethAssetId &&
+      avalancheAssetNames
+        .concat(optimismAssetNames)
+        .concat(bnbsmartchainAssetNames)
+        .includes(asset.name)
+    ) {
+      asset.name = `${asset.name} on Ethereum`
+    }
+
     // mark any avalanche assets that also exist on other evm chains
     if (
       chainReference === CHAIN_REFERENCE.AvalancheCChain &&
-      ethAssetNames.concat(optimismAssetNames).includes(asset.name)
+      asset.assetId !== avalancheAssetId &&
+      ethAssetNames.concat(optimismAssetNames).concat(bnbsmartchainAssetNames).includes(asset.name)
     ) {
       asset.name = `${asset.name} on Avalanche`
     }
 
+    // mark any bnbsmartchain assets that also exist on other evm chains
+    if (
+      chainReference === CHAIN_REFERENCE.BnbSmartChainMainnet &&
+      ethAssetNames.concat(optimismAssetNames).concat(avalancheAssetNames).includes(asset.name)
+    ) {
+      asset.name = `${asset.name} on BNB Smart Chain`
+    }
+
     // mark any optimism assets that also exist on other evm chains
     if (
       chainReference === CHAIN_REFERENCE.OptimismMainnet &&
-      ethAssetNames.concat(avalancheAssetNames).includes(asset.name)
+      ethAssetNames.concat(avalancheAssetNames).concat(bnbsmartchainAssetNames).includes(asset.name)
```

Comment:
As we add more EVM chains, we may want to make this programmatic, could basically take the full set of asset names in across all chains and filter out the current chain, just to make the logic more obvious, DRY, and make sure we get the logic right every single time e.g don't forget one when diffing

Reply:
https://github.com/shapeshift/lib/pull/1197/commits/2723dc3d169c7e25428aca3a4fe7a1865d774989